File: bindings/java-jni/src/main/java/org/maplibre/nativejni/map/ProjectionModeOptions.java
package org.maplibre.nativejni.map;

/** Mutable descriptor for axonometric map projection mode options. */
public final class ProjectionModeOptions {
  private Boolean axonometric;
  private Double xSkew;
  private Double ySkew;

  public boolean hasAxonometric() {
    return axonometric != null;
  }

  public Boolean axonometric() {
    return axonometric;
  }

  public ProjectionModeOptions axonometric(boolean axonometric) {
    this.axonometric = axonometric;
    return this;
  }

  public ProjectionModeOptions clearAxonometric() {
    axonometric = null;
    return this;
  }

  public boolean hasXSkew() {
    return xSkew != null;
  }

  public Double xSkew() {
    return xSkew;
  }

  public ProjectionModeOptions xSkew(double xSkew) {
    this.xSkew = xSkew;
    return this;
  }

  public ProjectionModeOptions clearXSkew() {
    xSkew = null;
    return this;
  }

  public boolean hasYSkew() {
    return ySkew != null;
  }

  public Double ySkew() {
    return ySkew;
  }

  public ProjectionModeOptions ySkew(double ySkew) {
    this.ySkew = ySkew;
    return this;
  }

  public ProjectionModeOptions clearYSkew() {
    ySkew = null;
    return this;
  }
}

File: bindings/kotlin-native/src/nativeMain/kotlin/org/maplibre/nativeffi/map/ProjectionModeOptions.kt
package org.maplibre.nativeffi.map

/** Mutable descriptor for axonometric map projection mode options. */
public class ProjectionModeOptions {
  public var axonometric: Boolean? = null
    private set

  public var xSkew: Double? = null
    private set

  public var ySkew: Double? = null
    private set

  public fun hasAxonometric(): Boolean = axonometric != null

  public fun axonometric(axonometric: Boolean): ProjectionModeOptions = apply {
    this.axonometric = axonometric
  }

  public fun clearAxonometric(): ProjectionModeOptions = apply { axonometric = null }

  public fun hasXSkew(): Boolean = xSkew != null

  public fun xSkew(xSkew: Double): ProjectionModeOptions = apply { this.xSkew = xSkew }

  public fun clearXSkew(): ProjectionModeOptions = apply { xSkew = null }

  public fun hasYSkew(): Boolean = ySkew != null

  public fun ySkew(ySkew: Double): ProjectionModeOptions = apply { this.ySkew = ySkew }

  public fun clearYSkew(): ProjectionModeOptions = apply { ySkew = null }
}
